RecoLocalCalo/HGCalRecProducers/python/HGCalRecHit_cfi.py

```diff
@@ -2,6 +2,60 @@
 from SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi import *
 from RecoLocalCalo.HGCalRecProducers.HGCalUncalibRecHit_cfi import *
 
+from Configuration.Eras.Modifier_phase2_hgcalV12_cff import phase2_hgcalV12
+
+def calcWeights(weightsPerLayer): res =  [sum(wei)/2. for wei in zip(weightsPerLayer[:], weightsPerLayer[1:] + [weightsPerLayer[0]])]; res[0] = 0.0; res[-1] = res[-2]; return res;   
```

Comment:
Would it make sense to call the first `weight` as `dummy_weight`, to make it clear that it is associated with a non-existing layer? 
That would also make the formula a little more explicit by using that value instead of `weightsPerLayer[0]`, to make it clear that you are also adding at the end a dummy value, not the first weight, that would make no sense.
The same would be true for `res[0] = dummy_weight` as well.
Actually, you could directly add `[weightsPerLayer[-1]]` as the last element and remove the `res[-1] = res[-2]`, to make it explicit that the last layer has no mean.

Reply:
I'm still convinced that extending the second array with the last element, and not with the first, dummy one, would make the code easier and remove the necessity of setting `res[-1]` manually, no?

Reply:
Ok, I removed it.